Mathlib/NumberTheory/NumberField/Units/Regulator.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2024 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import Mathlib.Algebra.Module.Zlattice.Covolume
+import Mathlib.LinearAlgebra.Matrix.Determinant.Misc
+import Mathlib.NumberTheory.NumberField.Units.DirichletTheorem
+
+/-!
+# Regulator of a number field
+
+We define and prove basic results about the regulator of a number field `K`.
+
+## Main definitions and results
+
+* `NumberField.Units.regulator`: the regulator of the number field `K`.
+
+* `Number.Field.Units.regulator_eq_det`: For any infinite place `w'`, the regulator is equal to
+the absolute value of the determinant of the matrix `(mult w * log w (fundSystem K i)))_i, w`
+where `w` runs through the infinite places distinct from `w'`.
+
+## Tags
+number field, units, regulator
+ -/
+
+open scoped NumberField
+
+noncomputable section
+
+namespace NumberField.Units
+
+variable (K : Type*) [Field K]
+
+open MeasureTheory Classical BigOperators NumberField.InfinitePlace
+  NumberField NumberField.Units.dirichletUnitTheorem
+
+variable [NumberField K]
+
+/-- The regulator of a number fied `K`. -/
+def regulator : ℝ := Zlattice.covolume (unitLattice K)
+
+theorem regulator_ne_zero : regulator K ≠ 0 := Zlattice.covolume_ne_zero (unitLattice K) volume
+
+theorem regulator_pos : 0 < regulator K := Zlattice.covolume_pos (unitLattice K) volume
+
+#adaptation_note
+/--
+After https://github.com/leanprover/lean4/pull/4119
+the `Module ℤ (Additive ((𝓞 K)ˣ ⧸ NumberField.Units.torsion K))` instance required below isn't found
+unless we use `set_option maxSynthPendingDepth 2`, or add
+explicit instances:
+```
+local instance : CommGroup (𝓞 K)ˣ := inferInstance
+```
+-/
+set_option maxSynthPendingDepth 2 -- Note this is active for the remainder of the file.
+
+theorem regulator_eq_det' (e : {w : InfinitePlace K // w ≠ w₀} ≃ Fin (rank K)) :
+    regulator K = |(Matrix.of fun i ↦ (logEmbedding K) (fundSystem K (e i))).det| := by
+  simp_rw [regulator, Zlattice.covolume_eq_det _
+    (((basisModTorsion K).map (logEmbeddingEquiv K)).reindex e.symm), Basis.coe_reindex,
+    Function.comp, Basis.map_apply, ← fundSystem_mk, Equiv.symm_symm]
+  rfl
+
+/-- Let `u : Fin (rank K) → (𝓞 K)ˣ` be a family of units. Then, for any infinite place `w'`, the
```

Comment:
In the docstring it seems you fix an infinite place `w'`, but in the statement of the theorem you fix two of them. Can you clarify things a little? Thanks!